sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -759,6 +759,8 @@ class Analyzer(
           u.failAnalysis(s"${ident.quoted} is a temp view not table.")
         }
         u
+      case u @ UnresolvedTableOrView(ident) =>
+        lookupTempView(ident).getOrElse(u)
```

Comment:
Looks like it is very similar to `UnresolvedRelation`?

`UnresolvedRelation` can also be resolve to view by `ResolveTempViews`, can be resolved to v2 relation by `ResolveTables`, can be resolved to v1 relation by `ResolveRelations`.

Seems we have a few similar ways to do resolving?

Reply:
some commands don't need to scan the table/view (like DESCRIBE). If we use `UnresolvedRelation`, then we will end up with `DataSourceV2ScanRelation`, which creates `Scan` from `Table` and is not necessary.

Another problem is view. `UnresolvedRelation` can be resolved to `View`, and then get removed at the beginning of optimizer.

I'll think of how to overcome these issues and reuse `UnresolvedRelation`.

Reply:
I did more investigation, and think we can't reuse `UnresolvedRelation`.

`UnresolvedRelation` means something we want to read/write, so it needs to resolve the data source, and fail if something goes wrong, e.g. the data source class can't be loaded.

For DESCRIBE commands, it should work as long as we can get the table. For example, we may want to use DESCRIBE to debug a table which can't load data source class.